stingray/lombscargle.py

```diff
@@ -0,0 +1,643 @@
+
+        lc2: :class:`stingray.lightcurve.Lightcurve` object
+            Another light curve to be Fourier transformed.
+            This is the reference band.
+
+        fullspec: boolean. Default is False.
+            If True, return the whole array of frequencies, or only positive frequencies (False).
+
+        method : str
+            The method to be used by the Lomb-Scargle Fourier Transformation function. `fast`
+            and `slow` are the allowed values. Default is `fast`. fast uses the optimized Press
+            and Rybicki O(n*log(n))
+
+        Returns
+        -------
+        freq: numpy.ndarray
+            The frequency grid at which the LSFT was evaluated
+
+        cross: numpy.ndarray
+            The cross spectrum value at each frequency.
+
+        """
+        if not freq:
+            freq = (
+                LombScargle(
+                    lc1.time,
+                    lc1.counts,
+                    fit_mean=False,
+                    center_data=False,
+                    normalization="psd",
+                ).autofrequency(
+                    minimum_frequency=max(self.min_freq, 0), maximum_frequency=self.max_freq
+                ),
+            )[0]
+            freqs2 = (
+                LombScargle(
+                    lc2.time,
+                    lc2.counts,
+                    fit_mean=False,
+                    center_data=False,
+                    normalization="psd",
+                ).autofrequency(
+                    minimum_frequency=max(self.min_freq, 0), maximum_frequency=self.max_freq
+                ),
+            )[0]
+            if max(freqs2) > max(freq):
+                freq = freqs2
+
+        if method == "slow":
+            lsft1 = lsft_slow(lc1.counts, lc1.time, freq)
+            lsft2 = lsft_slow(lc2.counts, lc2.time, freq)
+        elif method == "fast":
+            lsft1 = lsft_fast(lc1.counts, lc1.time, freq, oversampling=oversampling)
+            lsft2 = lsft_fast(lc2.counts, lc2.time, freq, oversampling=oversampling)
+        if fullspec:
+            lsft1, _ = impose_symmetry_lsft(lsft1, np.sum((lc1.counts)), lc1.n, freq)
+            lsft2, freq = impose_symmetry_lsft(lsft2, np.sum(lc2.counts), lc2.n, freq)
+        cross = np.multiply(lsft1, np.conjugate(lsft2))
+        return freq, cross
+
+    def _initialize_empty(self):
+        self.freq = None
+        self.power = None
+        self.power_err = None
+        self.unnorm_power = None
+        self.unnorm_power_err = None
+        self.df = None
+        self.dt = None
+        self.nphots1 = None
+        self.nphots2 = None
+        self.m = 1
+        self.n = None
+        self.fullspec = None
+        self.k = 1
+        self.err_dist = None
+        self.method = None
+        self.meancounts1 = None
+        self.meancounts2 = None
+        self.oversampling = None
+        self.variance1 = None
+        self.variance2 = None
+        return
+
+    def phase_lag(self):
```

Comment:
The test I added shows that this should work. At the moment, it only works if the time arrays are the same though. This should be understood. There is some reference phase I'm missing

Reply:
Yes. The tests are working, though have to relax the tolerances a tiny bit. 2% on absolute and relative each.